### PR TITLE
more aggressive reduction on import time

### PR DESCRIPTION
**Pull request by Jarvi-Izana**
since I have LazyCallable, now I can make recommender completely lazy-loaded, so does aggregate.

### CODE REVIEW

**Comment by Jarvi-Izana** [comment]
Hmmm, I branched from the previous import reduction branch and I merged that one. I thought it will diff based on the HEAD of the up-to-date master. I will fix it.

**Comment by Jarvi-Izana** [comment]
rebased; let's get it in.

File: src/python/turicreate/__init__.py [hunk 1]
```diff
@@ -19,9 +19,11 @@
 __version__ = '{{VERSION_STRING}}'
 from turicreate.version_info import __version__
 
-import turicreate.toolkits
 from turicreate.toolkits import evaluation
-import turicreate.aggregate
+# expose name aggregate as global variable
+# this won't interfere with import turicreate.aggregate as agg
```
Comment (TobyRoseman) [inline src/python/turicreate/__init__.py]: Does it interfere with tab complete in IPython?

File: src/python/turicreate/toolkits/recommender/__init__.py [hunk 2]
```diff
@@ -106,6 +108,7 @@
 item_similarity_recommender = _LazyModuleLoader(_mod_par + 'item_similarity_recommender')
 item_content_recommender = _LazyModuleLoader(_mod_par + 'item_content_recommender')
 
-# TODO: lazy callable
-from . import util
-from .util import _create as create
+# from . import util
```
Comment (TobyRoseman) [inline src/python/turicreate/toolkits/recommender/__init__.py]: Let's remove these commented out lines of code.

File: src/python/turicreate/__init__.py (hunk 1 above)
Comment (Jarvi-Izana) [reply src/python/turicreate/__init__.py]: nope.

File: src/python/turicreate/toolkits/recommender/__init__.py (hunk 2 above)
Comment (Jarvi-Izana) [reply src/python/turicreate/toolkits/recommender/__init__.py]: oh yeah

**Comment by Jarvi-Izana** [comment]
passed the internal tests. Let's get it in.

**Pull request merged**